packages/integration-tests/src/package_manager_sanity_checks.test.ts

```diff
@@ -81,76 +82,109 @@ void describe('getting started happy path', async () => {
-      windowsPathsNoEscape: true,
-      ignore: ['**/node_modules/**', '**/yarn.lock'],
-    });
-
-    const expectedAmplifyFiles = [
-      path.join('auth', 'resource.ts'),
-      'backend.ts',
-      path.join('data', 'resource.ts'),
-      'package.json',
-      'tsconfig.json',
-    ];
-
-    assert.deepStrictEqual(
-      files.sort(),
-      expectedAmplifyFiles.map((suffix) => path.join(pathPrefix, suffix))
-    );
+    await runWithRetry(
+      async () => {
+        if (packageManager === 'yarn-classic') {
+          await execa('yarn', ['add', 'create-amplify'], { cwd: tempDir });
+          await execaCommand(
+            './node_modules/.bin/create-amplify --yes --debug',
+            {
+              cwd: tempDir,
+              env: { npm_config_user_agent: 'yarn/1.22.21' },
+            }
+          );
+        } else {
+          await runPackageManager(
+            packageManager,
+            ['create', amplifyAtTag, '--yes'],
+            tempDir
+          ).run();
+        }
 
-    await runWithPackageManager(
-      packageManager,
-      [
-        'ampx',
-        'pipeline-deploy',
-        '--branch',
-        branchBackendIdentifier.name,
-        '--appId',
-        branchBackendIdentifier.namespace,
-      ],
-      tempDir,
-      { env: { CI: 'true' } }
-    ).run();
-
-    const clientConfigStats = await fsp.stat(
-      await getClientConfigPath(ClientConfigFileBaseName.DEFAULT, tempDir)
-    );
+        const pathPrefix = path.join(tempDir, 'amplify');
+
+        const files = await glob(path.join(pathPrefix, '**', '*'), {
+          nodir: true,
+          windowsPathsNoEscape: true,
+          ignore: ['**/node_modules/**', '**/yarn.lock'],
+        });
+
+        const expectedAmplifyFiles = [
+          path.join('auth', 'resource.ts'),
+          'backend.ts',
+          path.join('data', 'resource.ts'),
+          'package.json',
+          'tsconfig.json',
+        ];
 
-    assert.ok(clientConfigStats.isFile());
+        assert.deepStrictEqual(
+          files.sort(),
+          expectedAmplifyFiles.map((suffix) => path.join(pathPrefix, suffix))
+        );
+
+        await runWithPackageManager(
+          packageManager,
+          [
+            'ampx',
+            'pipeline-deploy',
+            '--branch',
+            branchBackendIdentifier.name,
+            '--appId',
+            branchBackendIdentifier.namespace,
+          ],
+          tempDir,
+          { env: { CI: 'true' } }
+        ).run();
+
+        const clientConfigStats = await fsp.stat(
+          await getClientConfigPath(ClientConfigFileBaseName.DEFAULT, tempDir)
+        );
+
+        assert.ok(clientConfigStats.isFile());
+      },
+      (error) => {
+        // Retry on network-related errors or command failures
+        return (
+          error.message.includes('exit code 1') &&
+            (error.message.includes('@aws-amplify/backend') ||
+              error.message.includes('aws-amplify')),
```

Comment:
```suggestion
            (error.message.includes('aws-amplify'),
```

there's some redundancy in this check.